### DISCUSSION TITLE
Improve LLM generated function summary in assertHE

### DISCUSSION THREAD

**Discussion by RobertASmith** [HTAhackathon2024/.github#15] [Ideas]
### What is the problem we're trying to solve

The [assertHE](https://github.com/dark-peak-analytics/assertHE) package currently includes the ability to generate a LLM summary of an R function. However, this may be biased by the function's documentation so we would ideally like to:
- adapt the code so that the prompt to the LLM includes more detailed information about just the model code, the documentation & the comments.
- ask the LLM to flag potential errors
- ask the LLM to write a test for the function?
- test the performance of the LLM on these tasks, how well does it perform - is the output acceptable?

### How will we tackle the problem?

To be decided, group discussion

### Which tools will we use?

- Existing package: https://github.com/dark-peak-analytics/assertHE
- OpenAI's GPT-3.5

### Previous work

- Issue here: https://github.com/dark-peak-analytics/assertHE/issues/69

**Comment by n8thangreen** [comment]
It would be great to learn how to use AI in `R`.

Could the AI generated test be linked to the unit test list I have for another project idea? It could perhaps act as an in-between to maybe adapt the inputs and outputs to match whats needed for the existing unit tests rather than write whole new tests?

**Reply by RobertASmith** [reply]
Yeah definitely. 

It could be that when creating a new test, the LLM is prompted to search through existing tests first before suggesting their own tests.
Finding the most efficient way to do this would be interesting.

**Comment by vikachooo** [comment]
Hi Rob,
I would really like to join this project. 
But I guess we would need an API key. If so, would you be willing to share one?

**Comment by RobertASmith** [comment]
Very pleased to have you join us.

To get a key please follow this guide:
https://github.com/dark-peak-analytics/assertHE/wiki/Using-the-LLM-functionality-in-assertHE

Look forward to seeing you tomorrow.

Kind regards,

Rob
